src/libcommon/log/sentry/handler.cpp

```diff
@@ -363,54 +371,58 @@ void SentryHandler::handleEventsRateLimit(SentryEvent &event, bool &toUpload) {
 }
 
-void SentryHandler::escalateSentryEvent(SentryEvent &event) {
+void Handler::escalateSentryEvent(SentryEvent &event) const {
     event.message += " (Rate limit reached: " + std::to_string(event.captureCount) + " captures.";
-    if (event.level == SentryLevel::Fatal || event.level == SentryLevel::Error) {
+    if (event.level == Level::Fatal || event.level == Level::Error) {
         event.message += ")";
         return;
     }
 
     event.message += " Level escalated from " + toString(event.level) + " to Error)";
-    event.level = SentryLevel::Error;
+    event.level = Level::Error;
 }
 
-void SentryHandler::updateEffectiveSentryUser(const SentryUser &user) {
+void Handler::updateEffectiveSentryUser(const SentryUser &user) {
     if (_globalConfidentialityLevel == _lastConfidentialityLevel && user.isDefault()) return;
     _lastConfidentialityLevel = user.isDefault() ? _globalConfidentialityLevel : SentryConfidentialityLevel::Specific;
 
     sentry_value_t userValue;
-    if (!user.isDefault()) {
+    if (_globalConfidentialityLevel == SentryConfidentialityLevel::Anonymous) {
+        userValue = toSentryValue(SentryUser("Anonymous", "Anonymous", "Anonymous"));
+        sentry_value_set_by_key(userValue, "ip_address", sentry_value_new_string("0.0.0.0"));
+        sentry_value_set_by_key(userValue, "authentication", sentry_value_new_string("Anonymous"));
+    } else if (!user.isDefault()) {
         userValue = toSentryValue(user);
         sentry_value_set_by_key(userValue, "ip_address", sentry_value_new_string("{{auto}}"));
         sentry_value_set_by_key(userValue, "authentication", sentry_value_new_string("Specific"));
     } else if (_globalConfidentialityLevel == SentryConfidentialityLevel::Authenticated) {
         userValue = toSentryValue(_authenticatedUser);
         sentry_value_set_by_key(userValue, "ip_address", sentry_value_new_string("{{auto}}"));
         sentry_value_set_by_key(userValue, "authentication", sentry_value_new_string("Authenticated"));
-    } else { // Anonymous
-        userValue = toSentryValue(SentryUser("Anonymous", "Anonymous", "Anonymous"));
-        sentry_value_set_by_key(userValue, "ip_address", sentry_value_new_string("-1.-1.-1.-1"));
-        sentry_value_set_by_key(userValue, "authentication", sentry_value_new_string("Anonymous"));
+    } else {
+        assert(false && "Invalid _globalConfidentialityLevel");
+        sentry_remove_user();
+        return;
     }
 
     sentry_remove_user();
     sentry_set_user(userValue);
 }
 
-void SentryHandler::writeEvent(const std::string &eventStr, bool crash) noexcept {
+void Handler::writeEvent(const std::string &eventStr, bool crash) noexcept {
     using namespace KDC::event_dump_files;
     auto eventFilePath =
-            std::filesystem::temp_directory_path() / (SentryHandler::appType() == AppType::Server
+            std::filesystem::temp_directory_path() / (Handler::appType() == AppType::Server
```

Comment:
This expression is quite complex!

Reply:
I think these are no longer used and remain from an attempt to fix the auto-restart behavior of the app.
@ChristopheLarchier, should we keep these functions?